client/my-sites/stats/stats-period-navigation/index.jsx

```diff
@@ -203,10 +206,55 @@ class StatsPeriodNavigation extends PureComponent {
 			<div
 				className={ clsx( 'stats-period-navigation', {
 					'stats-period-navigation__is-with-new-date-control': isWithNewDateControl,
+					'stats-period-navigation__is-with-new-date-filtering': isWithNewDateFiltering,
 				} ) }
 			>
 				<div className="stats-period-navigation__children">{ children }</div>
-				{ isWithNewDateControl ? (
+
+				{ /* Legacy view: Show only navigation arrows when not using new date control */ }
+				{ ! isWithNewDateControl && showArrows && (
```

Comment:
As the prop `isWithNewDateFiltering` is probably to be `true` only when `isWithNewDateControl` is true, we could make `isWithNewDateFiltering` as a feature flag check inside the `StatsPeriodNavigation` to simply the maintenance work in the future.

Reply:
Yes I definitely agree. I will resolve this in a follow up PR. For now I've addressed the other issues and will ship it 🚢